bindings/cs/MongoDB.Libmongocrypt/Binary.cs

```diff
@@ -65,14 +65,29 @@ public uint Length
         /// <summary>
         /// Converts to array.
         /// </summary>
-        /// <returns></returns>
         public byte[] ToArray()
         {
             byte[] arr = new byte[Length];
             Marshal.Copy(Data, arr, 0, arr.Length);
             return arr;
         }
 
+        /// <summary>
+        /// Write bytes into Data.
+        /// </summary>
+        public void WriteBytes(byte[] bytes)
+        {
+            if (Length == bytes.Length)
+            {
+                Marshal.Copy(bytes, 0, Data, (int)Length);
+            }
+            else
+            {
+                // this code path is not expected, but it's worth doing it to avoid silent saving of corrupted data
+                throw new InvalidDataException($"Incorrect bytes size. The bytes size must be {Length}.");
+            }
+        }
```

Comment:
Consider
```suggestion
        /// <summary>
        /// Write bytes into the data.
        /// </summary>
        public void WriteBytes(byte[] bytes)
        {
            if (bytes.Length != Length)
            {
                throw new InvalidDataException($"Inconsistent byte array size. Expected {Length} but found {bytes.Length}.");
            }
            
            Marshal.Copy(bytes, 0, Data, (int)Length);
        }
```

Reply:
idk, I don't mind about this code but at the same time don't see a reason to change the current one. So, I would leave the current code